types/react-dotdotdot/index.d.ts

```diff
@@ -0,0 +1,21 @@
+// Type definitions for react-dotdotdot 1.2
+// Project: https://github.com/CezaryDanielNowak/React-dotdotdot
+// Definitions by: Jakub Czyżewski <https://github.com/jczyzewski>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 2.8
+
+import { ComponentType } from 'react';
+
+interface DotDotDotProps {
+    clamp: number | string | boolean;
+    useNativeClamp?: boolean;
+    splitOnChars?: string[];
+    animate?: boolean;
+    truncationChar?: string;
+    truncationHTML?: string;
+    tagName?: string;
+}
+
+declare const Dotdotdot: ComponentType<DotDotDotProps>;
+
+export default Dotdotdot;
```

Comment:
This package was declared using `module.exports = ...` or some variant. It should be declared with an `export = ...` like one of the following:

### Variables

```ts
declare var xyz: SomeType;

export = xyz;
```

### Classes

```ts
class Foo {
  // ...
}

export = Foo
```

### Functions

```ts
function foo(): any;

export = foo;
```

To import the entity using a default import, you will need to enable `esModuleInterop`.

[See more here in the FAQ](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/9f9f6430117d025aef62ed56128deb65508edc86/README.md#a-package-uses-export--but-i-prefer-to-use-default-imports-can-i-change-export--to-export-default).